auth/src/main/java/org/openedx/auth/domain/interactor/MicrosoftSignInInteractor.kt

```diff
@@ -0,0 +1,76 @@
+package org.openedx.auth.domain.interactor
+
+import android.app.Activity
+import android.content.Context
+import com.google.gson.Gson
+import com.microsoft.identity.client.AcquireTokenParameters
+import com.microsoft.identity.client.AuthenticationCallback
+import com.microsoft.identity.client.IAuthenticationResult
+import com.microsoft.identity.client.PublicClientApplication
+import com.microsoft.identity.client.exception.MsalException
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import org.openedx.auth.data.model.SocialSighInStatus
+import org.openedx.core.BuildConfig
+import org.openedx.core.utils.Logger
+import java.io.File
+import java.net.URLEncoder
+
+class MicrosoftSignInInteractor(context: Context) {
+    private val logger = Logger(TAG)
+    private val microsoftConfig = mapOf(
+        "client_id" to BuildConfig.MICROSOFT_CLIENT_ID,
+        "redirect_uri" to "msauth://${context.packageName}/${
+            URLEncoder.encode(BuildConfig.MICROSOFT_PACKAGE_SIGN, "UTF-8")
+        }",
+        "account_mode" to "SINGLE",
+        "broker_redirect_uri_registered" to true
+    )
+
+    private val configJson = Gson().toJson(microsoftConfig)
+    private val configFile = File.createTempFile("config", null)
+        .also {
+            it.writeText(configJson)
+        }
```

Comment:
IMO, better to create the config file on build time.

Reply:
changed to gradle realisation